versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/DatabaseAdapter.java

```diff
@@ -323,4 +323,35 @@ void updateRepositoryDescription(Function<RepoDescription, RepoDescription> upda
    * @param offset initial reflog id to read from
    */
   Stream<RefLog> refLog(Hash offset) throws RefLogNotFoundException;
+
+  /**
+   * Retrieve the binary objects identified via {@code keys}. Binary objects that do not exist are
+   * not returned.
+   */
+  Stream<BinaryContent> getBinaries(Stream<BinaryKey> keys);
```

Comment:
Well, `globalContent` effectively returns opaque binary data too... Could we use a different name for these objects? "Binaries" seems to be too generic to me :thinking: